kuma/javascript/src/payments/formatDate.js

```diff
@@ -0,0 +1,44 @@
+// @flow
+
+/**
+ * Basic date formatting according to locale.
+ * Defaults to long month, numeric day and year
+ * e.g. for en-US, April 22, 2020
+ *
+ * @param {string} locale - locale identifier
+ * @param {Object | string | number} date - date to be formatted
+ * @param {Object} [options] - formatting options
+ * @returns {string} - formatted date string
+ */
+
+type DateTimeFormatOptions = {
+    localeMatcher?: 'lookup' | 'best fit',
+    timeZone?: string,
+    hour12?: boolean,
+    formatMatcher?: 'basic' | 'best fit',
+    weekday?: 'narrow' | 'short' | 'long',
+    era?: 'narrow' | 'short' | 'long',
+    year?: 'numeric' | '2-digit',
+    month?: 'numeric' | '2-digit' | 'narrow' | 'short' | 'long',
+    day?: 'numeric' | '2-digit',
+    hour?: 'numeric' | '2-digit',
+    minute?: 'numeric' | '2-digit',
+    second?: 'numeric' | '2-digit',
+    timeZoneName?: 'short' | 'long',
+};
+
+export const formatDate = (
+    locale: string,
+    date: Date | number | string,
+    options?: DateTimeFormatOptions = {
+        year: 'numeric',
+        month: 'long',
+        day: 'numeric',
+    }
+): string => {
+    const dateObj = new Date(date);
+    if (dateObj instanceof Date && isNaN(dateObj.valueOf())) {
+        return '';
```

Comment:
This is strange. If you passed in something so invalid/bad that it fails to make that type conversion you should probably throw an error instead of masking to an empty string.

Reply:
Thanks for the suggestion. I changed this so it throws an error instead.